src/connection.ts

```diff
@@ -25,48 +44,50 @@ import WS from '../websocket';
  * - state: Fired when the state of the Wallet changes.
  * - wallet-update: Fired when a new wallet message arrive from the websocket.
  **/
-class Connection extends EventEmitter {
+abstract class Connection extends EventEmitter {
+  // network: 'testnet' or 'mainnet'
+  public network: string;
+  public websocket: WalletWebSocket | WalletServiceWebSocket | null;
+  public currentServer: string;
+  public state: ConnectionState;
+
   /*
-   * network {String} 'testnet' or 'mainnet'
    * servers {Array} List of servers for the wallet to connect to, e.g. http://localhost:8080/v1a/
    */
-  constructor({
-    network,
-    servers = [],
-    connectionTimeout = null,
-  } = {}) {
+  constructor(options: ConnectionParams) {
     super();
 
+    const {
+      network,
+      servers,
```

Comment:
Not here, I'm handling `connectionTimeout` on `WalletServiceConnection` and `WalletConnection` because they instantiate the `WebSocket` differently on their constructors